corehq/apps/reports/standard/users/reports.py

```diff
@@ -0,0 +1,134 @@
+        'corehq.apps.reports.filters.users.ChangeActionFilter',
+        'corehq.apps.reports.filters.dates.DatespanFilter',
+    ]
+
+    description = ugettext_lazy("History of user updates")
+    ajax_pagination = True
+
+    sortable = False
+
+    @property
+    def default_datespan(self):
+        return datespan_from_beginning(self.domain_object, self.timezone)
+
+    @property
+    def headers(self):
+        h = [
+            DataTablesColumn(_("User")),
+            DataTablesColumn(_("By User")),
+            DataTablesColumn(_("Action")),
+            DataTablesColumn(_("Via")),
+            DataTablesColumn(_("Change Message")),
+            DataTablesColumn(_("Changes")),
+            DataTablesColumn(_("Timestamp")),
+        ]
+
+        return DataTablesHeader(*h)
+
+    @property
+    def total_records(self):
+        return self._get_queryset().count()
+
+    @memoized
+    def _get_queryset(self):
+        user_ids = self._get_user_ids()
+        changed_by_user_ids = self._get_changed_by_user_ids()
+        actions = self.request.GET.getlist('action')
+        query = self._build_query(user_ids, changed_by_user_ids, actions)
+        return query
+
+    def _get_user_ids(self):
+        es_query = self._get_users_es_query(self.request.GET.getlist(EMWF.slug))
+        return es_query.values_list('_id', flat=True)
+
+    def _get_changed_by_user_ids(self):
+        es_query = self._get_users_es_query(self.request.GET.getlist(ChangedByUserFilter.slug))
+        return es_query.values_list('_id', flat=True)
+
+    def _get_users_es_query(self, slugs):
+        return EMWF.user_es_query(
+            self.domain,
+            slugs,
+            self.request.couch_user,
+        )
+
+    def _build_query(self, user_ids, changed_by_user_ids, actions):
+        filters = Q(domain=self.domain)
+
+        if user_ids:
+            filters = filters & Q(user_id__in=user_ids)
+
+        if changed_by_user_ids:
+            filters = filters & Q(changed_by__in=changed_by_user_ids)
+
+        if actions and ChangeActionFilter.ALL not in actions:
+            filters = filters & Q(action__in=actions)
+
+        if self.datespan:
+            filters = filters & Q(changed_at__lt=self.datespan.enddate_adjusted,
+                                  changed_at__gte=self.datespan.startdate)
+        return UserHistory.objects.filter(filters)
+
+    @property
+    def rows(self):
+        records = self._get_queryset().order_by('-changed_at')[
+            self.pagination.start:self.pagination.start + self.pagination.count
+        ]
+        for record in records:
+            yield _user_history_row(record)
+
+
+def _user_history_row(record):
+    return [
+        cached_user_id_to_username(record.user_id),
+        cached_user_id_to_username(record.changed_by),
+        _get_action_display(record.action),
+        record.details['changed_via'],
+        record.message,
+        json.dumps(record.details['changes']),
```

Comment:
Agreed, specially the Create and Delete.

For the update ones, its still easy to comprehend since only a couple fields change.
There are couple of examples in the screenshot I have attached in description for reference.

I have done a raw dump right now to show all that we have, since product is going to revamp the UI anyway. So just avoiding any effort on UI for now.